src/context/user.tsx

```diff
@@ -0,0 +1,58 @@
+import { createContext, useState } from "react";
+import { IProfileUserProps } from "../types/profileUser";
+
+export const UserProfileContext = createContext({} as any);
```

Comment:
I know that at the beginning it's a good idea use the `any` to test the code. But don't forget of changing this to a specific type. If you use any you will not know which variables and methods you have access.

Reply:
Good evening man.

Was try resolve this trouble, but at assign a type like you suggest, i found a error. I Does'nt know like resolve. You can me send links like suggestion?

Change
---
![image](https://user-images.githubusercontent.com/58925056/220502279-289db7b9-9600-497e-9ff3-6ae6024a5b28.png)

Error after change
---
![image](https://user-images.githubusercontent.com/58925056/220502487-7f4ff241-a646-4ce5-bb7d-e2815e3a8633.png)

Reply:
In this Interface you can't type the children. You need to type the properties you want to share between the components.

Reply:
I made opposite, i assign type at provider not at context. Look under, worked

![image](https://user-images.githubusercontent.com/58925056/220641168-1de7cc99-b8a3-4673-98c4-fa4835254bcd.png)

Reply:
What I'm trying to say is that is a good idea to type the createContext({} as differentType), you know? Example:

```tsx
interface IUserProfileContextProvider {
   children: ReactNode | ReactNode[];
}

interface IUserProfileContextProps {
  // types of the variables you want to share between the components
}

export const UserProfileContext = createContext({} as IUserProfileContextProps );

export const UserProfileProvider = ({children}: IUserProfileContextProvider) => {
 // your states here
}

```